ext/fg/js/frontend-initialize.js

```diff
@@ -29,9 +31,32 @@ async function main() {
     const {id, depth=0, parentFrameId, ignoreNodes, url, proxy=false} = data;
 
     let popup;
-    if (proxy) {
+    if (!proxy && (window !== window.parent)) {
+        let rootPopupInformationResolve;
+        const rootPopupInformationPromise = new Promise((resolve) => (rootPopupInformationResolve = resolve));
+
+        const runtimeMessageCallback = ({action, params}, sender, callback) => {
+            if (action === 'rootPopupInformation') {
+                chrome.runtime.onMessage.removeListener(runtimeMessageCallback);
+                callback();
+                rootPopupInformationResolve(params);
+                return false;
+            }
+        };
+        chrome.runtime.onMessage.addListener(runtimeMessageCallback);
+        apiForward('rootPopupInformationGet');
+
+        const {popupId, frameId} = await rootPopupInformationPromise;
+
+        window._frameOffsetForwarder = new FrameOffsetForwarder();
```

Comment:
Any reason why this is assigned to `window._frameOffsetForwarder`?

Reply:
I was thinking that it would get garbage collected if there wasn't a reference to it anywhere (had this issue with Python earlier), but not sure how JS works. Also the instance created in the `else` block looked funny with just `new FrameOffsetForwarder();`. I could add a method that actually starts the listener to work around the issue while keeping the instance in local scope.

Reply:
It won't get GC'd because there are still refs to it in the event listeners.

```js
window.addEventListener('message', this.onMessage.bind(this), false);
```

This implicitly gives `window` a reference to `this`, since `this.onMessage` is bound.